src/main.rs

```diff
@@ -0,0 +1,8 @@
+use anyhow::Result;
+use didethresolver::run;
+
+#[tokio::main]
+async fn main() -> Result<()> {
```

Comment:
This separation of a main and lib is done to make unit testing possible

Reply:
Thank you!